### PR TITLE
Add width setting for the file finder

### PR DESCRIPTION
**Pull request by isaacdonaldson**
This PR adds the ability to adjust the width of the file finder popup. I found when searching my projects the default width was not always wide enough and there was no option to change it.

It allows values `small`, `medium` (default), `large`, `xlarge`, and `full`

Release Notes:

- Added a setting to adjust the width of the file finder modal

Example Setting:
```json
  "file_finder": {
    "modal_width": "medium"
  },
```

Screenshots can be found in the comments below.

### CODE REVIEW

**Comment by isaacdonaldson** [comment]
@cla-bot check

**Comment by thorbenk** [comment]
See https://github.com/zed-industries/zed/issues/16018

**Comment by isaacdonaldson** [comment]
> See https://github.com/zed-industries/zed/issues/16018

Ah my apologies, I looked for a past issue and couldn't find one, so I missed this one.

**Comment by isaacdonaldson** [comment]
> See #16018

Ah my apologies, I looked for a past issue and couldn't find one, so I missed this one. This PR would solve the issue of not being able to configure the width of the file finder that is mentioned there too.

**Comment by danilo-leal** [comment]
Very cool—thanks for the PR! I have a feeling it could make sense to tokenize the width value as opposed to being just an arbitrary number. We do already have `compact`, `comfortable`, and `default` as tokens that represent density (see https://github.com/zed-industries/zed/issues/18078), which we could possibly use here, too, but I'm not sure yet. And also, this doesn't necessarily mean that we wouldn't continue to allow passing arbitrary numbers—it's just so that there's a set of pre-established defaults that the whole app follows when it comes to UI config.

cc @iamnbutler for further thoughts!

**Comment by maxdeviant** [comment]
I think we may want to make the default behavior to have the modal size itself more intelligently to fit the available space rather than jumping to making the width configurable.

**Comment by isaacdonaldson** [comment]
> I think we may want to make the default behavior to have the modal size itself more intelligently to fit the available space rather than jumping to making the width configurable.

Yeah I agree. I changed it to have 4 width settings: `"small" | "medium" | "large" | "xlarge"` where `"medium"` is the default. The settings correspond to how much padding is on the side of the popup (`"small"` will have the largest padding). There is also a minimum width for the window so it will work fine on very small screens.

Screenshots:

Small:
Screenshot 2024-10-03 at 3 51 21 PM

Medium:
Screenshot 2024-10-03 at 3 54 12 PM

Large:
Screenshot 2024-10-03 at 3 52 34 PM

XLarge:
Screenshot 2024-10-03 at 3 52 55 PM


Here is the behaviour when the window is shrunk with the `"xlarge"` setting:


https://github.com/user-attachments/assets/019f9924-6e1c-4910-8b5d-d55e6f776ebe


And when it hits the minimum size:


https://github.com/user-attachments/assets/90d5fd8a-b723-4cbd-87a8-98d4088ff12c


And finally, what the `"xlarge"` setting looks like when in a small window:
Screenshot 2024-10-03 at 4 02 45 PM

**Comment by isaacdonaldson** [comment]
@alejandrovrojas - I added the `"full"` option for a fullscreen width finder window. 

Maybe someone from Zed can let me know if there are any desired changes, or if there is anything holding this PR up?

**Comment by thorbenk** [comment]
I tried this branch and I like it! It solves my problem https://github.com/zed-industries/zed/issues/16018

Thanks @isaacdonaldson, I hope this gets in :smile:

**Comment by protoEvangelion** [comment]
@maxdeviant @ConradIrwin @mrnugget  I'll literally switch from VSCode to Zed over this issue. It's been 3 years & they are still debating on how to allow users to override width: https://github.com/microsoft/vscode/pull/117862
Is there anything else that needs to be done on this PR to push it through?

**Review by SomeoneToIgnore — COMMENTED** [review]
This actually looks good, not sure why it got stalled.

Can we rebase and, optionally, add an entry to https://github.com/zed-industries/zed/blob/516f7b364213b2bbc1a658295df96443aa885ca0/docs/src/configuring-zed.md ?

And we're good to merge right after.

Edit: + Danilo's review comment is needed before the merge.

File: assets/settings/default.json
Comment (danilo-leal) [inline assets/settings/default.json:1]: This is awesome! Would it make more sense for the key to be `modal_width` instead of `window`? I feel like "window" refers to something else; something you'd be able to control, for example, via the app's window menu.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you!

**Comment by dangh** [comment]
It's might be too late to comment on this but I think computing the modal width by subtracting the paddings is not intuitive as it makes the modal "small" compare to the window, not the screen. On my mac, "small" modal is to wide on full screen width (takes too much of screen real estate and too wide compare to the command palette), and "medium" too small on half screen width (my paths are cut-off too much), which is very confusing.

Generally we want to define the minimum width of the modal so it can show long paths. If the modal is too wide for the small window, we shrink the modal so it can fit. An arbitrary number can help fine-tuned the setup for everyone.

Also a nitpicking but I would add small x-padding to the "full" modal as it has shadow and rounded corners, it would looks better this way.

**Pull request merged**
